index.js

```diff
@@ -3,53 +3,183 @@ const state = {
-      price: 0.35
+      price: 0.35,
     },
     {
       id: "010-eggplant",
       name: "eggplant",
-      price: 0.35
-    }
+      price: 0.35,
+    },
   ],
-  cart: []
-};
\ No newline at end of file
+  cart: [],
+};
+
+// for all the items in const state, render it as a li to replicates store-item.html in header (#id=store)
+// all the items should contain appropriate image and a button "add to cart" inside li
+const itemList = document.querySelector(".store--item-list");
+console.log(itemList);
+function calculateTotal() {
+  // this function should calculate total of the cart
+  let cartTotal = 0;
+  // calculate total
+
+  // display total
+  return cartTotal;
+}
+
+createItemList();
+function createItemList() {
+  state.items.forEach((item, index) => {
+    const li = document.createElement("li");
+    itemList.appendChild(li);
+
+    const div = document.createElement("div");
+    div.setAttribute("class", "store--item-icon");
+    li.appendChild(div);
+
+    const img = document.createElement("img");
+    img.setAttribute("src", `assets/icons/${item.id}.svg`);
+    img.setAttribute("alt", `${item.name}`);
+    div.appendChild(img);
+
+    const btn = document.createElement("button");
+    btn.innerText = "Add to cart";
+    li.appendChild(btn);
+
+    // when the add button is being pressed an event listener activate
+    // and adds the item in the item-list cart--item-list as a li
+    // render the item for user
+    // cart--item-list should contain image, name and quantity
+
+    const cartItem = document.querySelector(".cart--item-list");
+    btn.addEventListener("click", () => {
+      // store the item if the item is found in the cart
+
+      const itemFound = state.cart.find(
+        (itemFromCart) => itemFromCart.id === item.id
+      );
+
+      // i need to look at my cart if i already have item
+      // if i have then increase the quantity>
+
+      if (itemFound) {
+        itemFound.quantity++;
+      } else {
+        // if dont have it then run the following code
+        const addTocartItem = {};
+        addTocartItem.id = item.id;
+        addTocartItem.name = item.name;
+        addTocartItem.img = `assets/icons/${item.id}.svg`;
+        addTocartItem.quantity = 1;
+        state.cart.push(addTocartItem);
+
+        // rendering list of item in the cart
+        const li = document.createElement("li");
+        cartItem.appendChild(li);
+        state.items.push(li);
+
+        // rendering img in the cart
+        const img = document.createElement("img");
+        img.setAttribute("class", "cart--item-icon");
+        img.setAttribute("src", `assets/icons/${item.id}.svg`);
+        img.setAttribute("alt", `${item.name}`);
+        li.appendChild(img);
+
+        // rendering item name in the cart
+        const p = document.createElement("p");
+        p.innerText = ("p", `${item.name}`);
+        li.appendChild(p);
+
+        // rendering remove button element to decrease the quantity
```

Comment:
Pseudocode plus the code under it to do something is very nice, makes things easy to read.